### DISCUSSION TITLE
BUG: Error after RDP Login: Error code: 0x904 | RDP still works with Version 1.2.3

### DISCUSSION THREAD

**Discussion by Seamoconic** [rustdesk/rustdesk#10039] [Q&A] [answered]
I have a Problem with one PC.

When I connect to this PC via Rustdesk with RDP, I get the error message in the attachment after I have entered the login data of the PC. 
![image](https://github.com/user-attachments/assets/147d87ad-fcc6-4482-b276-304dbdfe517b)
Here again as text: 
Error code: 0x904
Extended error code: 0x0

Sometimes I can still see that it wants to log in, but most of the time the screen is simply black and then the error message appears. 

I only get the error message when I connect to this target PC. When I connect to other PC's with Rustdesk via RDP I get no error message and everything works fine. 

If I connect directly to this target PC via RDP without Rustdesk, it works without any problems. 

A second PC has the same hardware and the same OS (Win11). On the second PC, the connection with Rustdesk via RDP works. I compared the firewall settings but could not find any differences. 

I have started Rustdesk in debug mode but unfortunately there is no error message or anything similar. 

Does anyone have any idea what else I can check?

I tried that also but nothing worked:
Open the Registry Editor on the client (regedit.exe);
Go to the registry key HKEY_CURRENT_USER\Software\Microsoft\Terminal Server Client;
Create a DWORD parameter with the name RDGClientTransport and the value 1;
Restart the computer.

Open the local GPO editor and navigate to Computer Configuration > Administrative Templates > Windows Components > Remote Desktop Services > Remote Desktop Session Host > Security;
Enable the policy Require user of specific security layer for remote (RDP) connections. Select the RDP in Security Layer drop-down list. In this case, less secure native RDP encryption will be used to secure communications between the client and RD Session Host server.
Then enable the policy Set client connection encryption level. In the Encryption level drop-down box, select Client compliant. This will allow legacy RDP clients that don’t support 128-bit encryption to establish RDP connections to the host.

**Comment by enchuspe** [comment]
Hello,

I am experiencing an issue that is spreading across all my devices when using RustDesk to access via RDP. This error persists on both Windows 10 and Windows 11. Although I can establish RDP connections locally without any issues, attempting to do so through a tunnel with RustDesk makes the connection impossible.

Currently, I use Radmin VPN to connect via RDP to all my devices without any problems. I suspect that RustDesk has implemented some changes that might be affecting RDP connections. I have tried disabling the firewall, but this has not resolved the issue.

I have followed all available tutorials regarding error 0x904, but I have not been successful. I plan to try a version of RustDesk earlier than 1.3 and will share the results if I find a solution.

Best regards.

**Comment by Seamoconic** [comment]
Hello,

Thanks for the idea to check the versions. 

I have checked the latest versions. And indeed, the RDP function works again with version 1.2.3. 
Unfortunately, all versions that are newer do not work. 
Version 1.2.3 must be installed on the target PC. Newer versions can also be installed on the PC you want to connect to. 

@Rustdesk Can you please fix the RDP bug in newer versions.

Best regards
mathias

**Reply by enchuspe** [reply]
is correct, I installed version 1.2 and the connection via RDP works again. We could conclude then, that, the higher versions have some error that prevents to connect via RDP.
<img width="670" alt="image" src="https://github.com/user-attachments/assets/780504f7-27eb-45d3-8a36-e811cdbc9351">

@rustdesk

Best regards

**Comment by rustdesk** [comment]
@fufesou please test if we can reproduce this.

**Comment by rustdesk** [comment]
Please test the latest night build https://github.com/rustdesk/rustdesk/releases/tag/nightly
We have put a fix
You need to update the rustdesk client of controlled side.

**Reply by Seamoconic** [reply]
I carried out a test on myself. 
The following version has been installed on the target and source computers (both have Windows 11). 
https://github.com/rustdesk/rustdesk/releases/download/nightly/rustdesk-1.3.9-x86_64.exe
Was this the correct version?

Server hbbr and hbbs version is 1.1.14.

Unfortunately, I still get the following message
![image](https://github.com/user-attachments/assets/19c1a103-0e60-47c0-936e-843eab303f9b)

After I reinstalled the version rustdesk-1.2.3-2-x86_64.exe on the target computer, the connection via RDP worked again. 

Thank you for trying to solve the problem.

**Reply by fufesou** [reply, marked as answer]
@Seamoconic Thanks for your test.

> Was this the correct version?

No. You need to install https://github.com/rustdesk/rustdesk/releases/tag/nightly on the controlled side.

https://github.com/rustdesk/rustdesk/releases/download/nightly/rustdesk-1.4.0-x86_64.exe

**Reply by Seamoconic** [reply]
Perfect thank you, the connection works with this version :).
